### PR TITLE
Perform the opaque section variable inference outside of the kernel

### PR DESCRIPTION
**Pull request by ppedrot**
It is not the role of the kernel to decide to force the body of an entry to infer the section variable it uses, but the one of the upper layers. We make this explicit in the type of entries so as to enforce that this inference is performed beforehand.

Also removes auxilliary file stuff that doesn't look like it belongs in the kernel either.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Any reason not to do the same for transparent definitions?

**Comment by ppedrot** [comment]
This PR probably falls in the ballpark of @gares and @ejgallego.

**Comment by ejgallego** [comment]
> This PR probably falls in the ballpark of @gares and @ejgallego.

I'll be happy to assign this, and the PR looks OK to me, but IMHO this needs the OK of a kernel folk.

**Comment by ppedrot** [comment]
@SkySkimmer this could also be done, but the benefits are not as clear, and this requires more code refactoring in the upper layers. I am aiming for CEP 40, trying to get rid of all `Future.force` and `Future.join` from the kernel.

**Comment by ppedrot** [comment]
@SkySkimmer while you're here, you should look at the implementation of the section variable check in the kernel. This is fishy as hell, I am pretty sure we can find some soundness hole by poking strong enough via the ML API.

File: tactics/declare.ml
Comment (gares) [inline tactics/declare.ml]: I think it would be more clear to first match on `pure` and the `Future.force e.proof_entry_body` in both branches: in the first one not bind `eff` and but put `()` instead; in the second one bind `eff` and actually use it via `Safe_typing.push_private_constants`

Comment (gares) [reply tactics/declare.ml]: Also `ctx` is unused, isn't that fishy? If it is OK use `_ctx`

**Review by gares — CHANGES_REQUESTED** [review]
Looks good to me, modulo the changes I point out

File: tactics/declare.ml
Comment (ppedrot) [reply tactics/declare.ml]: Fixed.

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
ping @maximedenes

**Review by SkySkimmer — APPROVED** [review]
Looks nice, we even stop using aux_file in the kernel.

**Comment by SkySkimmer** [comment]
Nothing scary in here, I don't think we need to wait for @maximedenes to come back from holiday to merge.

**Comment by gares** [comment]
OK, but he is not on holidays, so this way of doing is totally unsound (for the future).

**Comment by ejgallego** [comment]
Indeed maybe we should have a coqdev-priv mailing list so people can say when they'll be offline etc...? @gares, Debian used to have something like that right?

**Comment by maximedenes** [comment]
> Nothing scary in here, I don't think we need to wait for @maximedenes to come back from holiday to merge.

As @gares said, I'm not on holidays, I'm having two busy weeks. Not sure what made you think it was the case.

**Comment by maximedenes** [comment]
I would have preferred that we put one more name in the kernel maintainers team (and create it as a github team), rather than having randomly (self-)selected assignees merge kernel patches on random assumptions.

**Comment by ejgallego** [comment]
> As @gares said, I'm not on holidays, I'm having two busy weeks. Not sure what made you think it was the case.

I guess this comment https://gitter.im/coq/coq?at=5d12562fce7f044dab65215c

But definitively it would be great we have a way to say "I'm away"

**Comment by Zimmi48** [comment]
> But definitively it would be great we have a way to say "I'm away"

It actually exists: GitHub now allows you to define a status, such as "on holiday" (possibly complemented with the busy checkbox to emphasize that you won't be reactive), and have this status be visible to everyone, or just the members of a single organization.

**Comment by Zimmi48** [comment]
> I would have preferred that we put one more name in the kernel maintainers team (and create it as a github team), rather than having randomly (self-)selected assignees merge kernel patches on random assumptions.

Would be definitely a good thing to do.

**Comment by ejgallego** [comment]
> or just the members of a single organization.

Oh, I ignored this existed, I discarded using Github due to privacy reasons. I am curious to know how the Debian system works tho.

**Comment by gares** [comment]
> It actually exists: GitHub now allows

We already have such a tool at Inria, it is called Zimbra, and it is filled in automatically when take holidays... if you don't look at it, then why should you look at the GH status (that one can forget to update, I surely would)?

My impression is that we could just take it easy, this PR was 3 days old...

**Comment by ejgallego** [comment]
> My impression is that we could just take it easy, this PR was 3 days old...

Hopefully we will discuss more on the next WG, but indeed something 3 days is just not manageable if you are trying to do some refactoring.

**Comment by gares** [comment]
> Hopefully we will discuss more on the next WG

Who is organizing it?

**Comment by ejgallego** [comment]
> Who is organizing it?

I could help with that indeed, I guess we want to do it in September, right?

**Comment by Zimmi48** [comment]
I almost mentioned the Inria Zimbra but refrained to do so because it is not a good way to make our development team open if we are assuming that we are all Inria employees (and as a matter of fact, it does not reflect today's reality even it might reflect tomorrow's).

**Pull request merged**
